### PR TITLE
Costume exchange NPC in malangdo

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: renewal

* **Description of Pull Request**: 

Content
-----------------------------
Initial implementation of the costume exchange NPC in malangdo converted from official script.
Several enchant stone and box have been added in item database to complete the script.

### CODE REVIEW

**Comment by Atemo** [comment]
@RagnaWay the raw just doesn't contain the kitcoin exchange menu. Do you know if the menu include it on KRO?

File: db/re/item_db.txt [hunk 1]
```diff
@@ -12110,11 +12205,52 @@
 25284,Swamp_Bug_Shell,Swamp Bug Shell,3,0,,10,,,,,,,,,,,,,{},{},{}
 25285,Brown_Rat_Tail,Brown Rat Tail,3,0,,10,,,,,,,,,,,,,{},{},{}
 25290,Sweets_Festival_Coin,Sweets Festival Coin,3,0,,0,,,,,,,,,,,,,{},{},{}
+25302,DoubleAttack_Stone,Double Attack Stone (Garment),3,20,,100,,,,,,,,,,,,,{ skill "TF_DOUBLE",3; bonus bDoubleRate,15; },{},{}
+25303,Critical_Stone_Robe,Critical Stone (Garment),3,20,,100,,,,,,,,,,,,,{ bonus bVariableCastrate,-20; },{},{}
```
Comment (peney7) [inline db/re/item_db.txt:12209]: shouldnt it be bCritAtkRate,3 ?

Comment (Atemo) [reply db/re/item_db.txt:12209]: You're right that is wrong but you're wrong about the rectification, it should be `bCritAtkRate,20`

File: db/re/item_db.txt [hunk 2]
```diff
@@ -12082,7 +12169,15 @@
 25184,Portable_Sewingbox,Portable Sewingbox,3,0,,0,,,,,,,,,,,,,{},{},{}
 25185,Locket_Pendant,Locket Pendant,3,0,,0,,,,,,,,,,,,,{},{},{}
 25187,Slug_Bullet,Slug Bullet,3,1200,,1200,,,,,,,,,,,,,{},{},{}
+25205,ShrinkEffect_Bottom,Shrink Effect (Lower),3,20,,100,,,,,,,,,,,,,{ /* todo */ },{},{}
+25206,ElectricEffect_Top,,3,20,,100,,,,,,,,,,,,,{ /* todo */ },{},{}
```
Comment (RagnaWay) [inline db/re/item_db.txt:12173]: Missing a column, Name

**Comment by teededung** [comment]
I think those enchant stones are wrong, need to remove all script bonus. Those items are just items to bring along for enhancement.

**Comment by Atemo** [comment]
Okay. Wait, what?

**Comment by teededung** [comment]
Yes, It's like that.
You can not enchant a stone that have weight. So all the stones has weight need to remove the script bonus.

For example: STR +1 with id 6636, when you bring this stone to npc to enchant costume, the costume will have STR +1 with id 4700

**Comment by teededung** [comment]
Some stones that I have filtered out long ago, I will share it for you to add it
```
4700, // STR +1
4710, // INT +1
4730, // AGI +1
4720, // DEX +1
4740, // VIT +1
4750, // LUK +1
4926, // CRIT +1
4930, // Healing Item +2%
4936, // Large Size +1% | Large Attack 1
4937, // Medium Size +1% | Medium Attack 1
4938, // Small Size +1% | Small Attack 1
29028, // ATK +1%(Upper)
29030, // MATK +1%(Upper)
4807, // ASPD +1
29053, // -1% delay skill | Delay After Skill1 Upper
29159, // +2% EXP | Experience +2%(Upper)
29156, // -3% VCT | Reduce Variable Cast(Upper)
29032, //  SP Absorp 1
29208, // SP Absorp 2
29013, // HP Absorp 1 3
29210, // HP Absorp 2 3
29359, // +3% CRIT damage | Fatal(Upper)

4882, // ATK +1%(Middle)
4883, // MATK +1%(Middle)
4927, // MaxHP +50
4931, // Recover HP +10 10s
4861, // MaxHP +1%
4929, // MaxSP +1%
29026, // DEF +20
29024, // LUK +3, STR -3
29014, // STR +3, INT -3
29022, // AGI +3, LUK -3
29016, // INT +3, DEX -3
29020, // VIT +3, AGI -3
29018, // DEX +3, VIT -3
29047, // +3% CRIT damage | Fatal(Middle)
29048, // +3% Long ranged damage | Expert Archer
29054, // Delay After Skill1 Middle
29145, // Experience +2%(Middle)
29157, // Reduce Variable Cast(Middle)
29155, // Autocast Lex Aeterna skill 

4884, // HIT +1
4859, // Flee +1
4928, // MaxSP +10
4932, // Recovery 1 SP when killed a monster

29021, // VIT +3, LUK -3
29023, // AGI +3, STR -3
29019, // DEX +3, AGI -3
29025, // LUK +3, INT -3
29015, // STR +3, DEX -3
29017, // INT +3, VIT -3
29033, // MDEF +4
29027, // Experience +2%(Lower)
29029, // ATK +1%(Lower)
29031, // MATK +1%(Lower)
29046, // Enable Greed skill
29049, // MaxHP +100, MaxHP +2% (Base Level 120)
29050, // MaxSP +50, MaxSP +2% (Base Levle 120)
29051, // Enable Detoxify skill
29052, // Enable Recovery skill
29057, // Enable Kyrie Eleison skill
29146, // Enable Item Appraisal skill
29147, // Enable Resurrection skill
29158, // Reduce Variable Cast(Lower)
29227, // Enable Heal skill
29229, // Enable Teleport skill
29228, // Enable Steal skill
29360, // +3% CRIT damage | Fatal(Lower)
29055; // -1% delay skill | Delay After Skill1 Lower
```

**Comment by saya9200** [comment]
is the broken english on those NPCs intended?

**Pull request merged**